### PR TITLE
add empty bucket method

### PR DESCRIPTION
**Pull request by mcalhoun**

### CODE REVIEW

File: modules/aws/s3.go [hunk 1]
```diff
@@ -139,6 +139,72 @@ func DeleteS3BucketE(t *testing.T, region string, name string) error {
...
+		}
+
+		//Running the Bulk delete job (limit 1000)
+		_, err = s3Client.DeleteObjects(deleteParams)
+		if err != nil {
+			return err
+		}
+
+		if *(*bucketObjects).IsTruncated { //if there are more objects in the bucket, IsTruncated = true
+			params.Marker = (*deleteParams).Delete.Objects[len((*deleteParams).Delete.Objects)-1].Key
```
Comment (brikis98) [inline modules/aws/s3.go]: `bucketObjects.Marker` should do this in a much simpler manner :)

File: modules/aws/s3.go [hunk 2]
```diff
@@ -139,6 +139,72 @@ func DeleteS3BucketE(t *testing.T, region string, name string) error {
...
+			logger.Logf(t, "Requesting next batch | %s", *(params.Marker))
```
Comment (brikis98) [inline modules/aws/s3.go]: Do you need to sleep in between to avoid being throttled?

**Comment by brikis98** [comment]
Also, does this work if the bucket has versioning enabled?

File: modules/aws/s3.go (hunk 2 above)
Comment (mcalhoun) [reply modules/aws/s3.go]: It's actually a bulk delete so only 1 operation, so I don't think it will be throttled.

**Comment by mcalhoun** [comment]
Ok, this has been updated to include tests which upload a random number of files to the bucket with both versioned and non-versioned buckets.

**Pull request merged**